lib/matplotlib/ticker.py

```diff
@@ -1159,11 +1161,18 @@ def _non_decade_format(self, sign_string, base, fx, usetex):
         if is_close_to_int(coeff):
             coeff = round(coeff)
         if usetex:
-            return (r'$%s%g\times%s^{%d}$') % \
-                                        (sign_string, coeff, base, exponent)
+            return ((r'$%s%g\times%s^{%d}$') %
+                    (sign_string, coeff, base, exponent))
         else:
             return ('$%s$' % _mathdefault(r'%s%g\times%s^{%d}' %
-                                        (sign_string, coeff, base, exponent)))
+                    (sign_string, coeff, base, exponent)))
+
+
+class SymmetricalLogFormatter(LogFormatterSciNotation):
```

Comment:
Sorry to be a pain but... should we also have SymLogFormatter / SymLogFormatterExponent / SymLogFormatterMathtext / SymLogFormatterSciNotation?  Or does that point out to the fact that we should refactor the various LogFormatter subclasses into kwargs to a single class first?  Or am I just overthinking this?

Reply:
More thought about this is welcome! I think on the most basic level the formatters boil down to 3 cases:

- Everything scalar formatted
- Everything log formatted
- A mix of the above (ie. `SymLogFormatter`)

and slight variations have grown on top of the basic log formatter. I'm not sure what the ideal way of mixing these together is though when it comes to `SymLog`, will have a think.